src/server/game/Handlers/NPCHandler.cpp

```diff
@@ -391,12 +391,12 @@ void WorldSession::SendSpiritResurrect()
         if (corpseGrave != ghostGrave)
             _player->TeleportTo(corpseGrave->map_id, corpseGrave->x, corpseGrave->y, corpseGrave->z, _player->GetOrientation());
         // or update at original position
-        else
-            _player->UpdateObjectVisibility();
+        //else
+        //    _player->UpdateObjectVisibility(); called in ResurrectPlayer
```

Comment:
It is called in ResurrectPlayer so absolutely not needed, ill apply all suggestions if someone decides it's worth merging.